presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -255,7 +256,7 @@ public void retainedBytesForEachPart(BiConsumer<Object, Long> consumer)
     {
         consumer.accept(dictionary, dictionary.getRetainedSizeInBytes());
         consumer.accept(ids, sizeOf(ids));
-        consumer.accept(this, (long) INSTANCE_SIZE);
+        consumer.accept(this, (long) INSTANCE_SIZE + DICTIONARY_ID_SIZE);
```

Comment:
After discussing with @highker  and @haozhun, we think for this simple case, let's just add the overhead of `DictionaryId` into the `INSTANCE_SIZE` :). I will merge it once Jenkins is cleared.